### DISCUSSION TITLE
return www-authenticate with 401 from auth server

### DISCUSSION THREAD

**Discussion by tykling** [bluenviron/mediamtx#1117] [Questions]
Hello, how come the server doesn't add a www-authenticate: header when returning 401 to HTTP clients (for HLS)?
401 with no www-authenticate: header means the browser doesn't prompt for credentials or send the ones embedded in the url.
Thanks!

**Comment by aler9** [comment]
Hello, the server does add the `WWW-Authenticate` header to 401 responses:

https://github.com/aler9/rtsp-simple-server/blob/27fb9fdb7ea53f8d13a0fda2a6a59b361f87a4c3/internal/core/hls_muxer.go#L448

Can you post instructions on how to replicate the issue?

**Comment by tykling** [comment]
hello :) thank you for your response!

My config enables auth with an external auth server. It works well for RTMPS and RTSPS, the auth server is contacted when a client connects, and if my auth api returns 200 the stream client is permitted. So far so good.

The issue is with HLS where it seems like the browser is not sending credentials. I assumed this was because the HLS server was responding with 401 without a `www-authenticate` header, but maybe I am just doing it wrong.

Let me ask a different way, how do I watch the HLS stream produced by `rtsp-simple-server `when I have auth enabled? How do I provide the credentials? Do I need to do anything special to get firefox or chrome to show the HLS stream? What other clients are known to work?

I tried with no credentials expecting a normal HTTP auth popup from the browser, but none came. Then I tried embedding credentials like `https://user:pass@host:8888/streamname` but the browser doesn't send the credentials it seems. I then tried https://user:pass@host:8888/streamname with curl and that actually worked (I got back some HTML and js), although I can't watch a stream with curl :)

I am clearly doing something wrong, but what? thanks! :)

**Reply by aler9** [reply]
I discovered that, in case of external authentication, the WWW-Authentication header is not sent. I fixed the issue some days ago: https://github.com/aler9/rtsp-simple-server/commit/1e788f81fd46c7e8f5314bd4ae04989debfff52c

**Reply by tykling** [reply]
oh very nice! Then it wasn't just me :) 
Do you want me to convert this discussion to an issue, with a reference to the commit, so it can easily be found by others?
Thanks!

**Reply by tykling** [reply]
Also, is there a release planned with this fix soon, or is there alternatively a recommended way of getting a docker image with the fix?

[a comment by an automated account is not shown]
